src/Service/Startup.cs

```diff
@@ -189,7 +190,15 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<IAuthorizationResolver, AuthorizationResolver>();
             services.AddSingleton<IOpenApiDocumentor, OpenApiDocumentor>();
 
-            AddGraphQLService(services);
+            int? depthLimit = null;
+            if (runtimeConfig is not null
+                && runtimeConfig.Runtime!.GraphQL!.DepthLimit is not null
+                && runtimeConfig.Runtime!.GraphQL!.DepthLimit > 0)
+            {
+                depthLimit = runtimeConfig.Runtime!.GraphQL!.DepthLimit;
+            }
+
+            AddGraphQLService(services, depthLimit);
```

Comment:
would it make more sense to pass in the object type `GraphQLRuntimeOptions` here? 
+ we don't have to change method signature later if/when we want to pass more config options to GraphQL service setup.
+ then your logic for parsing out DepthLimit can exist in one place close to where it is used -> `AddGraphQLService(...)` where all GraphQL service config is set.

Reply:
Yes, thats a good point. thanks for highlighting that.